src/app/(nextjs_migration)/components/client/PageLoadEvent.tsx

```diff
@@ -23,13 +23,57 @@ export const PageLoadEvent = (props: Props) => {
   const { pageEvents } = useGlean(props.channel, props.appEnv);
   const pathname = usePathname();
 
+  let url = "";
+  let referrer = "";
+  let utm_campaign = "";
+  let utm_content = "";
+  let utm_medium = "";
+  let utm_source = "";
+  let utm_term = "";
+
+  if (
+    typeof window !== "undefined" &&
+    typeof document !== "undefined" &&
+    window.location &&
+    document
+  ) {
+    url = new URL(pathname, window.location.origin).toString();
+    referrer = document.referrer;
+
+    const params = new URLSearchParams(window.location.search);
+    utm_campaign = params.get("utm_campaign") ?? "";
+    utm_content = params.get("utm_content") ?? "";
+    utm_medium = params.get("utm_medium") ?? "";
+    utm_source = params.get("utm_source") ?? "";
+    utm_term = params.get("utm_term") ?? "";
```

Comment:
Q: is this used? I don't see it below, only the first 4 (campaign, content, medium, source).

Reply:
Oops good catch yes I missed `term`.